tests/test_compat.py

```diff
@@ -34,3 +36,32 @@ def test_raises_StructError_on_overflow(self):
 
     def test_byte_literal(self):
         self.assertIsInstance(b('abc'), bytes)
+
+
+class TestBytes(unittest.TestCase):
+    """Tests for bytes objects."""
+
+    def test_xor_bytes(self):
+        b1 = b'\xff\xff\xff\xff'
+        b2 = b'\x00\x00\x00\x00'
+        b3 = b'\xf0\xf0\xf0\xf0'
```

Comment:
Test with some random (but still hard-coded) byte strings. It's too easy to miss subtle errors when you only test with all-on or all-off values.